lms/services/blackboard_api/client.py

```diff
@@ -27,20 +27,28 @@ def get_token(self, authorization_code):
         """
         self._api.get_token(authorization_code)
 
-    def list_files(self, course_id):
-        """Return the list of files in the given course."""
+    def list_files(self, course_id, folder_id=None):
+        """Return the list of files in the given course or folder."""
 
-        files = []
-        path = f"courses/uuid:{course_id}/resources?type=file&limit={PAGINATION_LIMIT}"
+        path = f"courses/uuid:{course_id}/resources"
+
+        if folder_id:
+            # Get the files and folders in the given folder instead of the
+            # course's top-level files and folders.
+            path += f"/{folder_id}/children"
+
+        path += f"?limit={PAGINATION_LIMIT}"
+
+        results = []
 
         for _ in range(PAGINATION_MAX_REQUESTS):
```

Comment:
Not necessarily for this PR but I think we should move pagination logic to BasicClient and take a param or similar on `_api.request`

Reply:
Yep. I think we'll wait until we actually have a second method that uses pagination, then we'll be able to see what sort of generic design will work